### PR TITLE
add CwmsUtility class for converting between DSS paths and CWMS time-series-id's

### PR DESCRIPTION
**Pull request by ktarbet**
Merged readme and readme2

### CODE REVIEW

File: src/hecdss/cwms_utility.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        tsid = ""
+        p = DssPath(path, 0)
+        a, b, c, e, f = p.A.strip(), p.B.strip(), p.C.strip(), p.E.strip(), p.F.strip()
+        # [Location]
+        if a != "":
+            # [Location]
+            tsid += a[:24] + "-"  # limit base location to 24 characters
+        if b != "":
+            # [sub-location]
+            tsid += b[:32]   # limit sub-location to 32 characters
```
Comment (perrymanmd) [inline src/hecdss/cwms_utility.py:85]: I've never seen The A and B parts interpreted as location and sub-location. The A part (the watershed by convention) is ignored completely for CWMS and the full location name comes from the B part.

File: src/hecdss/cwms_utility.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        tsid += "." + c + "."
+        # [Type]
+        tsid += CwmsUtility.get_cwms_parameter_type(dss_type) + "."
+        # [Interval]
+        tsid += CwmsUtility._convert_dss_interval_to_cwms_interval(e) + "."
+        # [Duration]
+        tsid += duration+"."
+        # [Version]
+        tsid += f
+
```
Comment (perrymanmd) [inline src/hecdss/cwms_utility.py:97]: CWMS Convention is title case for all the parts so I would call the `title()` function on all of the parts that are copied from the pathname.

**Review by oskarhurst — APPROVED** [review]

File: src/hecdss/dss_type.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+class DssType(Enum):
+    PER_AVER = "PER-AVER"
+    PER_CUM = "PER-CUM"
+    INST_VAL = "INST-VAL"
+    INST_CUM = "INST-CUM"
+    FREQ = "FREQ"
+    PER_MAX = "PER-MAX"
+    PER_MIN = "PER-MIN"
+    CONST = "CONST"
+
```
Comment (perrymanmd) [inline src/hecdss/dss_type.py:13]: Only the first 4 are currently DSS standards, but you know I'm open to expanding this!
https://www.hec.usace.army.mil/confluence/dssdocs/dssjavaprogrammer/time-series-data/units-and-type

**Review by perrymanmd — COMMENTED** [review]
Some comment for consideration, but I won't get in the way of you getting this ready for the workshop.

File: Readme.md [hunk 4]
```diff
@@ -1,8 +1,110 @@
...
+# Open a DSS file
+file_path = "example.dss"
+dss = DssFile(file_path)
+
+# Retrieve and print data
+data_path = "/example/data/////"
+data = dss.get(data_path)
+print(data)
+
+dss.values = dss.values * 2
```
[the remark this answers is not in the document]
Comment (ktarbet) [reply Readme.md:53]: Thanks, will change

File: src/hecdss/cwms_utility.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        '1Minute': '1Minute',
+        '2Minutes': '2Minute',
+        '3Minutes': '3Minute',
+        '4Minutes': '4Minute',
+        '5Minutes': '5Minute',
+        '6Minutes': '6Minute',
+        '12Minutes': '12Minute',
+        '15Minutes': '15Minute',
+        '20Minutes': '20Minute',
+        '30Minutes': '30Minutes',
```
[the remark this answers is not in the document]
Comment (ktarbet) [reply src/hecdss/cwms_utility.py:18]: good catch. Thanks

File: src/hecdss/cwms_utility.py [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+        # # DSS doesn't support the 2Day, 3Day, etc...  intervals.
+        # '2Days': '',
+        # '3Days': '',
+        # '4Days': '',
+        # '5Days': '',
+        # '6Days': '',
+        '1Week': '1Week',
+        '1Month': '1Month',
+        '1Year': '1Year',
+        '0': 'Ir-Year'  # not optimized for some cases
```
[the remark this answers is not in the document]
Comment (ktarbet) [reply src/hecdss/cwms_utility.py]: done

File: src/hecdss/dss_type.py (hunk 3 above)
Comment (ktarbet) [reply src/hecdss/dss_type.py:13]: I'll annotate the file as such

File: src/hecdss/cwms_utility.py (hunk 1 above)
Comment (ktarbet) [reply src/hecdss/cwms_utility.py:85]: I'll drop the A part, and It looks like all the paths in test_dsspath.py will still pass.

I found that sub-location technique in some Java code.  DSSPathname.pathnameToTsId(), but let's go with the drop-watershed convention.

File: src/hecdss/cwms_utility.py (hunk 2 above)
Comment (ktarbet) [reply src/hecdss/cwms_utility.py]: I added an optional strict option , that calls title().  

```python
def pathname_to_cwms_tsid(path, dss_type="", duration="0", strict=False):
```

**Pull request merged**